catroid/src/org/catrobat/catroid/content/bricks/AskBrick.java

```diff
@@ -0,0 +1,266 @@
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.View.OnTouchListener;
+import android.widget.AdapterView;
+import android.widget.AdapterView.OnItemSelectedListener;
+import android.widget.BaseAdapter;
+import android.widget.CompoundButton;
+import android.widget.CompoundButton.OnCheckedChangeListener;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.InternToExternGenerator;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.ui.adapter.DataAdapter;
+import org.catrobat.catroid.ui.adapter.UserVariableAdapterWrapper;
+import org.catrobat.catroid.ui.dialogs.NewDataDialog;
+import org.catrobat.catroid.ui.fragment.FormulaEditorFragment;
+
+import java.util.List;
+
+public class AskBrick extends UserVariableBrick {
+
+	private static final long serialVersionUID = 1L;
+	private transient AdapterView<?> adapterView;
+
+	private transient String defaultPrototypeToken = null;
+
+	public AskBrick(Formula questionFormula, UserVariable answerVariable) {
+		this.userVariable = answerVariable;
+		initializeBrickFields(questionFormula);
+	}
+
+	public AskBrick(String questionText) {
+		this.userVariable = null;
+		initializeBrickFields(new Formula(questionText));
+	}
+
+	private void initializeBrickFields(Formula questionFormula) {
+		addAllowedBrickField(BrickField.ASK_QUESTION);
+		setFormulaWithBrickField(BrickField.ASK_QUESTION, questionFormula);
+	}
+
+	@Override
+	public int getRequiredResources() {
+		return getFormulaWithBrickField(BrickField.ASK_QUESTION).getRequiredResources();
+	}
+
+	@Override
+	public List<SequenceAction> addActionToSequence(Sprite sprite, SequenceAction sequence) {
+		sequence.addAction(sprite.getActionFactory().createAskAction(sprite,
+				getFormulaWithBrickField(BrickField.ASK_QUESTION), userVariable));
+		return null;
```

Comment:
I suggest returning `Collections.emptyList()` as the preferred way if the returned list is not modified